cxx/distributions/normal.hh

```diff
@@ -37,19 +59,52 @@ public:
         var -= (x - mean) * (x - old_mean);
     }
 
+    void posterior_hypers(double *mprime, double *sprime) {
+      // r' = r + N
+      // m' = (r m + N mean) / (r + N)
+      // C = N (var + mean^2)
+      // s' = s + C + r m^2 - r' m' m'
+      double mdelta = r * (m - mean) / (r + N);
+      *mprime = mean + mdelta;
+      *sprime = s + r * (m * m - *mprime * *mprime)
+          + N * (var - 2 * mean * mdelta - mdelta * mdelta);
+    }
+
     double logp(double x) const {
-        double y = (x - mean);
-        return -0.5 * (y * y / var + log(var) + log(M_2PI));
+      // Based on equation (13) of GaussianInverseGamma.pdf
+      double unused_mprime, sprime;
+      incorporate(x);
```

Comment:
I think this will hit an error because logp is declared const? (This is an issue that also came up in #7. This pattern of incorporating and unincorporating makes sense to me, but I'm not sure what the best way to implement it in C++ is. Maybe we get rid of the `const` declaration?)

Reply:
You are correct.  Fixed it by const_casting this.